### PR TITLE
Uprev cryptfs-tpm2 to support both tpm2-tss 1.x and 2.x

### PR DESCRIPTION
**Pull request by yunguowei**
@jiazhang0 

This is v2 with adding tss2 1.x support and changes for v1 review comment.

Yunguo

### CODE REVIEW

File: src/include/cryptfs_tpm2.h [hunk 1]
```diff
@@ -163,7 +247,7 @@ extern const char *cryptfs_tpm2_build_machine;
 extern int option_quite;
 extern bool option_no_da;
 
-#define TPM_ALG_AUTO		0x4000
+#define TPM2_ALG_AUTO		0x4000
```
Comment (jiazhang0) [inline src/include/cryptfs_tpm2.h:256]: Everything looks good, except this macro should be enclosed in TSS2_LEGACY_V1 and !TSS2_LEGACY_V1.

**Review by jiazhang0 — APPROVED** [review]

File: src/lib/tcti.c [hunk 2]
```diff
@@ -110,26 +141,38 @@ init_tcti_device(void)
 TSS2_TCTI_CONTEXT *
 init_tcti_socket(void)
 {
+#ifndef TSS2_LEGACY_V1
+	const char *cfg = "host=127.0.0.1,port=2321";
```
Comment (jiazhang0) [inline src/lib/tcti.c:145]: Is there a macro defining the port number and local address in a tss2 header?

File: NEWS [hunk 3]
```diff
@@ -1,5 +1,11 @@
-Version 0.6.2
+Version 0.7.0
 =============
+Released: 2018-09-06
+
+- Rebased to tpm2-tss 2.0
+
+=============
+Version 0.6.2
```
Comment (jiazhang0) [inline NEWS:8]: Should be:
```
Version 0.6.2
=========
```

**Review by jiazhang0 — CHANGES_REQUESTED** [review]

File: scripts/luks-setup.sh [hunk 4]
```diff
@@ -100,6 +100,9 @@ detect_tpm() {
 
         grep -q "TPM 2.0 Device" "/sys/class/tpm/$dev/device/description" 2>/dev/null &&
             tpm_absent=0 && break
+
+        grep -q "TPM 2.0 Device" "/sys/class/tpm/$dev/device/firmware_node/description" 2>/dev/null &&
```
Comment (jiazhang0) [inline scripts/luks-setup.sh:104]: Please add a comment describing the reason as the commit header mentioned.

File: src/lib/session.c [hunk 5]
```diff
@@ -62,6 +80,8 @@ set_session_auth(TPMS_AUTH_COMMAND *session, TPMI_SH_AUTH_SESSION handle,
 		       session->hmac.t.size);
 	} else
 		session->hmac.t.size = 0;
+
```
Comment (jiazhang0) [inline src/lib/session.c:83]: Remove this empty line please.

**Comment by jiazhang0** [comment]
Most of patches look good. Send a V3 PR please.

File: src/include/cryptfs_tpm2.h (hunk 1 above)
Comment (yunguowei) [reply src/include/cryptfs_tpm2.h:256]: TPM_ALG_AUTO was local macro, not from tpm2-tss. I just renamed it to TPM2_ALG_AUTO, all references to it are TPM2_ALG_AUTO, no matter it is v1 or v2 , and no TPM_ALG_AUTO anymore. So I think it is good.

File: src/lib/tcti.c (hunk 2 above)
Comment (yunguowei) [reply src/lib/tcti.c:145]: I was trying to find the macro, but unfornately did find it. 

Some sample code in tpm2-tss are all hard-coded:

    expect_value(__wrap_Tss2_Tcti_Mssim_Init, tctiContext, NULL);
    expect_any(__wrap_Tss2_Tcti_Mssim_Init, size);
    expect_string(__wrap_Tss2_Tcti_Mssim_Init, config, "host=localhost,port=2321");
    will_return(__wrap_Tss2_Tcti_Mssim_Init, lsize);
    will_return(__wrap_Tss2_Tcti_Mssim_Init, TSS2_RC_SUCCESS);

    expect_any(__wrap_Tss2_Tcti_Mssim_Init, tctiContext);
    expect_memory(__wrap_Tss2_Tcti_Mssim_Init, size, &lsize, sizeof(lsize));
    expect_string(__wrap_Tss2_Tcti_Mssim_Init, config, "host=localhost,port=2321");
    will_return(__wrap_Tss2_Tcti_Mssim_Init, lsize);

**Comment by yunguowei** [comment]
I will send v3 soon.

**Pull request closed without merging**